Makefile

```diff
@@ -166,7 +166,10 @@ eunit: export COUCHDB_QUERY_SERVER_JAVASCRIPT = $(shell pwd)/bin/couchjs $(shell
 eunit: couch
 	@$(REBAR) setup_eunit 2> /dev/null
 	@for dir in $(subdirs); do \
-	  $(REBAR) -r eunit $(EUNIT_OPTS) apps=$$dir; \
+	  $(REBAR) -r eunit $(EUNIT_OPTS) apps=$$dir;  \
+          if grep -r 'context setup failed' -C1 src/$$dir/.eunit/couch.log ; then \
```

Comment:
What if couch.log doesn't get created? I'm seeing the following output while testing a new ken PR:
```
$ rm -rf src/ken/.eunit && make eunit apps=ken
==> ken (compile)
Compiled src/ken_server.erl
==> rel (compile)
==> couchdb (compile)
==> couchdb (setup_eunit)
==> ken (eunit)
Compiled src/ken_app.erl
Compiled src/ken_sup.erl
Compiled src/ken.erl
Compiled src/ken_event_handler.erl
/Users/jay/proj/couchdb/src/ken/test/ken_server_test.erl:15: Warning: export_all flag enabled - all functions will be exported
Compiled test/ken_server_test.erl
Compiled src/ken_server.erl
======================== EUnit ========================
module 'ken'
module 'ken_event_handler'
ken_server: prune_old_entried_test (module 'ken_server')...[0.308 s] ok
module 'ken_sup'
module 'ken_app'
module 'ken_server_test'
  undefined
  *** context setup failed ***
**in function ken_server_test:setup_default/0 (test/ken_server_test.erl, line 65)
**error:{badmatch,{error,{badarg,[{ets,new,
                               [ken_workers,[named_table,public,{keypos,2}]],
                               []},
                          {ken_server,init,1,
                                      [{file,"src/ken_server.erl"},
                                       {line,118}]},
                          {gen_server,init_it,2,
                                      [{file,"gen_server.erl"},{line,374}]},
                          {gen_server,init_it,6,
                                      [{file,"gen_server.erl"},{line,342}]},
                          {proc_lib,init_p_do_apply,3,
                                    [{file,"proc_lib.erl"},{line,249}]}]}}}


undefined
*** context setup failed ***
**in function ken_server_test:setup_default/0 (test/ken_server_test.erl, line 63)
**error:{badmatch,{error,{already_started,<0.172.0>}}}


undefined
*** context setup failed ***
**in function ken_server_test:setup_default/0 (test/ken_server_test.erl, line 63)
**error:{badmatch,{error,{already_started,<0.172.0>}}}


undefined
*** context setup failed ***
**in function ken_server_test:setup_default/0 (test/ken_server_test.erl, line 63)
**error:{badmatch,{error,{already_started,<0.172.0>}}}


undefined
*** context setup failed ***
**in function ken_server_test:setup_default/0 (test/ken_server_test.erl, line 63)
**error:{badmatch,{error,{already_started,<0.172.0>}}}


undefined
*** context setup failed ***
**in function ken_server_test:setup_default/0 (test/ken_server_test.erl, line 63)
**error:{badmatch,{error,{already_started,<0.172.0>}}}


undefined
*** context setup failed ***
**in function ken_server_test:setup_default/0 (test/ken_server_test.erl, line 63)
**error:{badmatch,{error,{already_started,<0.172.0>}}}


undefined
*** context setup failed ***
**in function ken_server_test:setup_default/0 (test/ken_server_test.erl, line 63)
**error:{badmatch,{error,{already_started,<0.172.0>}}}


undefined
*** context setup failed ***
**in function ken_server_test:setup_default/0 (test/ken_server_test.erl, line 63)
**error:{badmatch,{error,{already_started,<0.172.0>}}}


undefined
*** context setup failed ***
**in function ken_server_test:setup_config/0 (test/ken_server_test.erl, line 71)
**error:{badmatch,{error,{already_started,<0.172.0>}}}


[done in 0.011 s]
=======================================================
  Failed: 0.  Skipped: 0.  Passed: 1.
One or more tests were cancelled.
ERROR: One or more eunit tests failed.
ERROR: eunit failed while processing /Users/jay/proj/couchdb/src/ken: rebar_abort
grep: src/ken/.eunit/couch.log: No such file or directory
jay@mbp:~/proj/couchdb
$ echo $?
0
```

Reply:
Then this scheme might not work. It seems context setup might not end up there at all. I forgot that eunit.log is a couch_log log not EUnit's own output log.